sdk/eas/src/eas.ts

```diff
@@ -1,98 +1,482 @@
-import { SchemaRegistry } from "@ethereum-attestation-service/eas-sdk";
-import { validate } from "@settlemint/sdk-utils/validation";
-import { getPublicClient, getWalletClient } from "@settlemint/sdk-viem";
-import type { PublicClient } from "viem";
-import { type ClientOptions, ClientOptionsSchema } from "./client-options.schema.js";
-import { publicClientToProvider, walletClientToSigner } from "./ethers-adapter.js";
-import type { RegisterSchemaOptions } from "./types.js";
-import { buildSchemaString, validateSchemaFields } from "./validation.js";
-
-// Re-export types and constants
-export type { ClientOptions, ClientOptionsSchema } from "./client-options.schema.js";
-export type { RegisterSchemaOptions, SchemaField, EASFieldType } from "./types.js";
-export { EAS_FIELD_TYPES } from "./types.js";
+import { createPortalClient } from "@settlemint/sdk-portal";
+import type { Address, Hex } from "viem";
+import { GraphQLOperations } from "./graphql/operations.js";
+import { ZERO_ADDRESS } from "./types.js";
+import type {
+  AttestationInfo,
+  AttestationRequest,
+  DeploymentResult,
+  EASClientOptions,
+  GetAttestationsOptions,
+  GetSchemasOptions,
+  PortalTransactionResponse,
+  SchemaData,
+  SchemaField,
+  SchemaRequest,
+  TransactionResult,
+} from "./types.js";
 
 /**
- * Creates an EAS client for interacting with the Ethereum Attestation Service.
- *
- * @param options - Configuration options for the client
- * @returns An object containing the EAS client instance
- * @throws Will throw an error if the options fail validation
- *
- * @example
- * ```ts
- * import { createEASClient } from '@settlemint/sdk-eas';
- *
- * const client = createEASClient({
- *   schemaRegistryAddress: "0x1234567890123456789012345678901234567890",
- *   attestationAddress: "0x1234567890123456789012345678901234567890",
- *   accessToken: "your-access-token",
- *   chainId: "1",
- *   chainName: "Ethereum",
- *   rpcUrl: "http://localhost:8545"
- * });
- * ```
+ * Main EAS client class for interacting with Ethereum Attestation Service via Portal
```

Comment:
Keep the nice jsDoc codeblocks, these will be in the generated documentation (run `bun run docs` in the root of the project to see the final output in the main README.md)